src/engine/cachingreader.cpp

```diff
@@ -204,12 +204,17 @@ CachingReaderChunkForOwner* CachingReader::lookupChunkAndFreshen(SINT chunkIndex
 void CachingReader::newTrack(TrackPointer pTrack) {
     auto newState = pTrack ? STATE_TRACK_LOADING : STATE_TRACK_UNLOADING;
     auto oldState = m_state.fetchAndStoreAcquire(newState);
-    VERIFY_OR_DEBUG_ASSERT(
-            oldState != STATE_TRACK_LOADING ||
+
+    // TODO():
+    // BaseTrackPlayerImpl::slotLoadTrack() distributes the new track via
+    // emit(loadingTrack(pNewTrack, pOldTrack));
+    // but the newTrack may change if we load a new track while the previous one
+    // is still loading. This leads to inconsistent states for example a different
+    // track in the Mixx Title and the Deck label.
+    if (oldState != STATE_TRACK_LOADING ||
```

Comment:
Condition needs to be inverted: == && ==

Reply:
Ups .. yes!